sdk/src/main/scala/com/horizen/account/validation/ChainIdBlockSemanticValidator.scala

```diff
@@ -0,0 +1,26 @@
+package com.horizen.account.validation
+
+import com.horizen.account.block.AccountBlock
+import com.horizen.account.transaction.{AccountTransactionsIdsEnum, EthereumTransaction}
+import com.horizen.params.NetworkParams
+import com.horizen.validation.SemanticBlockValidator
+
+import scala.util.Try
+
+case class ChainIdBlockSemanticValidator(params: NetworkParams) extends SemanticBlockValidator[AccountBlock] {
+  override def validate(block: AccountBlock): Try[Unit] = Try {
+    for (actx <- block.transactions) {
```

Comment:
Wouldn't it be better moving this validation to `EthereumTransaction.semanticValidity()`?

After all this is more a tx validation, and the predefined block validator (`SidechainBlockSemanticValidator`) already loops on txes and calls that method for each tx.